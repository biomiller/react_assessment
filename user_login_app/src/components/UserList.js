import React from 'react';
import { User } from './User.js';


export function UserList(props) {

    return (
        <div>
            <h1>Users:</h1>
            {props.data.map(user => (
                <div key={user._id}>
                    <br></br>
                    <User
                        user={user}
                    />
                </div>
            ))
            }
        </div>

    );
}