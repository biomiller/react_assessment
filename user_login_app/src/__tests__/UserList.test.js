import React from 'react';
import renderer from 'react-test-renderer';

import { MemoryRouter } from 'react-router-dom';

import { UserList } from "../components/UserList.js";


it('is UserList rendered', () => {

    const data=[{username:"Test"}]

    const component = renderer
    .create(
      <MemoryRouter>
        <UserList data={data} />
      </MemoryRouter>
    )
    .toJSON();

    expect(component).toMatchSnapshot();
});



