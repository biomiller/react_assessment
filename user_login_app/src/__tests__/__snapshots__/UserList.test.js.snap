// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is UserList rendered 1`] = `
<div>
  <h1>
    Users:
  </h1>
  <div>
    <br />
    <div>
      <h3>
        Test
      </h3>
    </div>
  </div>
</div>
`;
